src/__test__/Media.test.tsx

```diff
@@ -530,6 +530,46 @@ describe("Media", () => {
     })
   })
 
+  describe("prevent nested unnecessary renders", () => {
+    it("only renders one element when Media is nested within Media", () => {
+      const query = renderer.create(
+        <MediaContextProvider>
+          <Media at="extra-small">
+            <span className="extra-small" />
+          </Media>
+
+          <Media at="medium">
+            <Media at="extra-small">
+              <span className="extra-small" />
+            </Media>
+            <Media at="medium">
+              <span className="medium" />
+            </Media>
+            <Media at="large">
+              <span className="large" />
+            </Media>
+          </Media>
+
+          <Media at="large">
+            <span className="large" />
+          </Media>
+        </MediaContextProvider>
+      )
+
+      expect(
+        query.root.findAllByProps({ className: "extra-small" }, { deep: true })
+          .length
+      ).toBe(1)
+      expect(
+        query.root.findAllByProps({ className: "large" }, { deep: true }).length
+      ).toBe(1)
+      expect(
+        query.root.findAllByProps({ className: "medium" }, { deep: true })
+          .length
+      ).toBe(1)
```

Comment:
It’s been a while since I’ve been in this codebase, but wouldn’t we expect 2 `div` elements with the `medium` class? Or perhaps I missed the part in the `Media` component that would omit rendering a second nested `div`?

Reply:
Ah, this is because the test finds where `className=medium`, the divs rendered within won't have the class name, hence the total should be 1.

Reply:
Ah yes of course, we’re asserting for the existence of the span 👍